pkg/provisioning/config/yaml/linter.go

```diff
@@ -16,62 +16,52 @@ package yaml
 
 import (
 	"context"
-	"fmt"
 	"sort"
 
-	"github.com/Masterminds/semver/v3"
+	"github.com/conduitio/conduit/pkg/foundation/cerrors"
 	"github.com/conduitio/conduit/pkg/foundation/log"
 	"github.com/conduitio/conduit/pkg/provisioning/config/yaml/internal"
 	"github.com/conduitio/yaml/v3"
 )
 
 type configLinter struct {
-	changelog internal.Changelog
-
 	expandedChangelog map[string]map[string]any
-	version           string
-	warnings          warnings
 }
 
-func newConfigLinter(changelog internal.Changelog) *configLinter {
-	cl := &configLinter{
-		changelog: changelog,
+func newConfigLinter(changelogs ...internal.Changelog) *configLinter {
+	// expand changelogs
+	expandedChangelog := make(map[string]map[string]any)
+	for _, changelog := range changelogs {
+		for k, v := range changelog.Expand() {
+			if _, ok := expandedChangelog[k]; ok {
+				panic(cerrors.Errorf("changelog already contains version %v", k))
+			}
+			expandedChangelog[k] = v
+		}
 	}
-	cl.init()
-	return cl
-}
-
-func (cl *configLinter) init() {
-	cl.expandedChangelog = cl.changelog.Expand()
 
-	var versions semver.Collection
-	for k := range cl.changelog {
-		versions = append(versions, semver.MustParse(k))
+	return &configLinter{
+		expandedChangelog: expandedChangelog,
 	}
-	sort.Sort(versions)
-
-	// latest version is the default version
-	cl.version = versions[len(versions)-1].Original()
 }
 
-func (cl *configLinter) InspectNode(path []string, node *yaml.Node) {
-	if len(path) == 1 && path[0] == "version" {
```

Comment:
The linter doesn't have to figure out the version itself anymore. We parse the version before actually decoding the config since we need to choose the types based on the version, so we supply it to the linter as well.